### ISSUE TITLE
Outline rendering glitches in wasm

### ISSUE THREAD

**Issue by luca-della-vedova** [komadori/bevy_mod_outline#19]
Hi and thanks for the great crate!

We are using the crate for our [application](https://github.com/open-rmf/rmf_site), we insert a `Flat` `SetOutlineDepth` component and show the outline when the mouse is hovering.
While this looks great on a desktop (Linux, AMD64, integrated GPU with Vulkan) build:

[desktop_view.webm](https://github.com/komadori/bevy_mod_outline/assets/9111558/b154b6ba-f623-4902-8de9-5c758e5ff567)

It looks very odd on wasm:

[wasm_view.webm](https://github.com/komadori/bevy_mod_outline/assets/9111558/bb033266-a8fc-46a8-ba08-d5028e725c6d)

The code is exactly the same, do you have any idea what might be happening?

**Comment by komadori** [comment]
The effect you're seeing is Z-fighting between the outline and the stencil. Since the fix for #14, bevy_mod_outline uses a depth offset to separate the outline and stencil. For some reason, the required separation is much larger in WebGL than the native back-ends. I'm still investigating.

**Comment by komadori** [comment]
The issue is, due to https://github.com/gfx-rs/naga/issues/2362, the floating point error in the depth values is much bigger with the WebGL backend than non-OpenGL backends, requiring a much bigger depth bias. I've added a GL-specific workaround to bevy_mod_outline which avoids the increased floating point error at the cost of some performance.

Released in 0.4.2.

**Comment by luca-della-vedova** [comment]
Wow I had a feeling the fix was tricky but didn't think it would require this depth of digging.
Thanks a lot for looking into it and releasing the fix so quickly!

**Comment by dannymcgee** [comment]
Hi there! Unfortunately it seems like this issue has cropped up again (or maybe my particular hardware setup is just immune to the previous fix). I forked this locally and was able to fix the issue by increasing the offset to a cartoonishly large amount, which doesn't seem to have introduced any negative side effects. (I can email you before & after screenshots if that would be helpful.)

This is the change I made: 
https://github.com/komadori/bevy_mod_outline/blob/3740766157062c99f685510833f06aa520045469/src/outline.wgsl#L111

```wgsl
    out.normalized_depth = 10.0 + 0.5 * (out_z / clip_pos.w);
```

Would be happy to open a PR if you'd like to incorporate it, but no hard feelings if you'd prefer to try a more robust solution or wait for an upstream fix from wgpu.
